core/src/main/java/com/datastax/oss/driver/internal/core/metadata/NodeInfo.java

```diff
@@ -107,20 +118,24 @@
    * relies on this information). If you're not using token metadata in any way, you may return an
    * empty set here.
    */
+  @Nullable
   Set<String> getTokens();
 
   /**
    * An additional map of free-form properties, that can be used by custom implementations. They
    * will be copied as-is into {@link Node#getExtras()}.
    */
+  @Nullable
   Map<String, Object> getExtras();
 
   /**
    * The host ID that is assigned to this host by cassandra. The driver uses this to uniquely
    * identify a node.
    */
+  @NonNull
```

Comment:
So I double-checked this and this needs to be non-null, otherwise the driver will throw NPEs in many places. I changed the peer check logic in `DefaultTopologyMonitor` to include the broadcast RPC address and the host ID.